activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -20,6 +20,22 @@ def write_query?(sql) # :nodoc:
           !READ_QUERY.match?(sql.b)
         end
 
+        LOCK_QUERY = ActiveRecord::ConnectionAdapters::AbstractAdapter.build_lock_query_regexp(
+          # table-level locks
+          :lock_table, :lock_tables, :unlock_table, :unlock_tables,
+          # row-level locks
+          :for_update, :for_share,
+          # read lock for table flushing
+          :read_lock
+        ) # :nodoc:
+        private_constant :LOCK_QUERY
+
+        def lock_query?(sql) # :nodoc:
+          LOCK_QUERY.match?(sql)
```

Comment:
Not a big fan of having to match one more regexp, especially such a large one, on every queries.

If anything I recently tried to not need to match any regexp on queries by passing down whether we're doing a read or a write, and only fallback to regexp when the query has an unknown origin.

So I get the want for this feature, but it is one more step in the direction opposite to the one I'm trying to go :/

Reply:
I think it would be worth exploring indeed.

When I get time I'll try to finalize my attempt of no longer matching regexp on the happy path, once it's done perhaps this feature will be easier to plug in? Unclear.